### DISCUSSION TITLE
How do I activate Dhizuku

### DISCUSSION THREAD

**Discussion by iamr0s** [iamr0s/Dhizuku#19] [Announcements]
English | [简体中文](https://github.com/iamr0s/Dhizuku/discussions/16)

# How do I activate Dhizuku

## List of possible problems

1. [Not allowed to set the device owner because there are already some `accounts` on the device.](#already_accounts)
2. [Not allowed to set the device owner because there are already some `users` on the device.](#already_users)
3. [Trying to set the device owner, but device owner is already set.](#already_device_owner)
4. [@ProvisioningPreCondition 99.](#oppo_sign)
5. [Shizuku option not finding](#shizuku)

### <a name="already_accounts">Existing account</a>
You must delete all accounts when activating Dhizuku.
You can uninstall or freeze applications with accounts (Freeze includes three ways: disable, hide, and suspend).
> [Hail](https://github.com/aistra0528/Hail) ([Shizuku](https://shizuku.rikka.app/) - Disable) is recommended here.

1. Find apps that may have accounts
Download and install the [Accounts](https://github.com/iamr0s/AndroidAccounts/releases), then click the top right button to export to the clipboard.
<image src="https://github.com/iamr0s/Dhizuku/assets/41138521/7fa00030-4489-4507-8b18-5a92a4de76e3" height="400"/>

2. Disable the apps
Download and install the [Hail](https://github.com/aistra0528/Hail/releases), then set the work mode to `Shizuku - Disable` .
<image src="https://github.com/iamr0s/Dhizuku/assets/41138521/8c108904-14e2-4ecd-83a6-d82b94683ef8" height="400"/>

3. Import the apps from clipboard
> You can also add it manually
<image src="https://github.com/iamr0s/Dhizuku/assets/41138521/f2b90ab5-127f-4061-b969-bea1f48dd527" height="400"/>

4. Freeze
> Click the freeze button to freeze, remember to unfreeze after activation or before restart, otherwise it may cause failure to boot.
<image src="https://github.com/iamr0s/Dhizuku/assets/41138521/60cd257a-51eb-46ad-8e2f-2595a3a5eeeb" height="400"/>

### <a name="already_users">Existing user</a>
No other user can exist while activating Dhizuku.
Users are typically created for app clone, double app, guest, and multi-user functions, there are several solutions:

- [Recommended] Deletes a user from Settings
The deletion entry of different custom systems is different, and this is for reference only.
![ed8b95237a8968ee3f14dee66459ead6](https://github.com/iamr0s/Dhizuku/assets/41138521/1c14235d-4c93-48bf-bf94-dc9dcac829fc)

- Run the ADB command to delete it

1. Querying a User ID
Command: `pm list users`
Output:
```
:/ $ pm list users
Users:
        UserInfo{0:Owner:c13} running
        UserInfo{999:XSpace:801010} running
```
The user ID of 'Owner' is'`0`, the user ID of the double-open space, which is `XSpace`, is `999`.

2. Remove a user
Command: `pm remove-user id`
Notice: You only need to remove users other than user ID `0`
Warning: Applications in the removed user space will be uninstalled
Output:
```
:/ $ pm remove-user 999
Success
```

- Do not delete the user
This method no longer works on some devices, so you can try it if you want to try your luck.
Command: `dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver`
Principle: Add the ID of the primary space to the original activation command, that is, limit the permission scope of the device owner to the primary space.
Output:
```
:/ $ dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver
Success: Active admin and device owner set to com.rosan.dhizuku/.server.DhizukuDAReceiver for user 0
```

### <a name="already_device_owner">The device owner already exists</a>
- Click on Dhizuku's forced stop and re-enter to see if Dhizuku has been activated
- Transfer device owner authority to Dhizuku
- Deactivate non-Dhizuku device owner applications

### <a name="oppo_sign">OPPO signature verification</a>
OPPO allows only whitelisted applications and test signature applications to activate the device owner.
So you can use [MT Manager](https://www.coolapk.com/apk/bin.mt.plus) to sign Dhizuku, then install and activate it again.
![image](https://github.com/iamr0s/Dhizuku/assets/41138521/0ac6c259-e842-4a70-9624-c8a594f4a925)
![40eaadb46fe6b8e868b89b63c8742938](https://github.com/iamr0s/Dhizuku/assets/41138521/3ad6fc2f-be70-4f16-afc2-bec32241a156)

### <a name="shizuku">Shizuku option not finding</a>

First, please [activate **Shizuku**](https://shizuku.rikka.app/guide/setup/#start-shizuku).

### <a name="download_links">Download Links</a>
- <a name="download_link_dhizuku">[Dhizuku](https://github.com/iamr0s/Dhizuku/releases)</a>

**Comment by GaneshGorana** [comment]
help me, i wanted to deactivate dhishuku app from option from this app, and now my device stuck and i can't click on screen any more , only my side fingerprint is working and nothing is happening, what should i do, please tell me

#edit
After 30 min approx, my device return normal.  Developer should fix that kind of bug

**Comment by backtick82** [comment]
what i did:
1. install and activate [shizuku](https://shizuku.rikka.app/)
2. install [hail](https://github.com/aistra0528/Hail) and set 'Working mode' to 'Shizuku - Disable'
3. authorize hail
4. use [Accounts](https://github.com/iamr0s/AndroidAccounts/releases) to get any apps that i needed
5. import them into hail, and freeze them
6. run `adb shell dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver` on my computer
7. unfreeze the apps, and remove them from the freezing list
8. open hail and set 'Working mode' to 'Dhizuku - Hide' or 'Dhizuku - Suspend'
9. add apps i need hidden or suspended
10. Done!

**Reply by Evengard** [reply]
Yeah, a friendly reminder, rebooting with Google Services disable may make your phone soft bricked, so there's that.
Make sure to authorize your phone on ADB before rebooting to have a chance at recovery

**Comment by Xisrr1** [comment]
Says there are accounts on the device even when there are none.

**Reply by AsterionXeity** [reply]
Same 
![Screenshot_20250818_070845](https://github.com/user-attachments/assets/6e678780-1da2-4435-b4a3-cf129b5f2332)
![Screenshot_20250818_070911](https://github.com/user-attachments/assets/d4971565-a8a4-4827-9eac-1ddd26631d2e)

**Reply by wilson2533** [reply]
Telegram or Whatsapp etc

**Reply by nktnet1** [reply]
Even if all accounts are removed, some applications may still be holding references to them.

You need to freeze/kill/uninstall those applications. Try running this for starters:

```sh
adb shell dumpsys account list
```

**Reply by ghyujkilop** [reply]
I tried this, and all the apps it showed in my list are already frozen. But still stuck.

**Reply by nktnet1** [reply]
@ghyujkilop what's your exact output of the command?

```
adb shell dumpsys account list
```

You should see:

```
User UserInfo{0:User:4c13}:
  Accounts: 0
  
  ...
  
  Active Sessions: 0
```

The typical cause is disabled apps still holding onto references to accounts (which won't be listed in your account settings.

More details here:

- https://github.com/BinTianqi/OwnDroid/issues/233#issuecomment-4564641131

**Comment by Daepikgoose** [comment]


**Reply by s1204IT** [reply]
> Not allowed to set the device owner because there are already some accounts on the device.

Try the "**Run the ADB command to delete it**" part.

**Reply by Xisrr1** [reply]
What command? I'm facing the same issue. Even after disabling all apps with accounts, it will still give this error.

**Reply by Brokoli5191** [reply]
if you removed all account from your device as described above, take the sim card out of your phone and reboot your device, then run the command again.

**Reply by extremecarver** [reply]
S24 Ultra, same problem. I removed the e-sim by actually removing/deleting it. Then I rebooted. I then disabled all apps which appear under account until all accounts are not showing up anymore. Tried setting owner - get denied. 
Rebooted just to be sure but still same problem:

C:\adb>adb shell
e3q:/ $ pm list users
Users:
        UserInfo{0:Username:4c13} running
e3q:/ $ exit

C:\adb>adb shell dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver

Exception occurred while executing 'set-device-owner':
java.lang.IllegalStateException: Not allowed to set the device owner because there are already some accounts on the device.
        at com.android.server.devicepolicy.DevicePolicyManagerService.enforceCanSetDeviceOwnerLocked(DevicePolicyManagerService.java:11852)
        at com.android.server.devicepolicy.DevicePolicyManagerService.setDeviceOwner(DevicePolicyManagerService.java:10454)
        at com.android.server.devicepolicy.DevicePolicyManagerServiceShellCommand.runSetDeviceOwner(DevicePolicyManagerServiceShellCommand.java:257)
        at com.android.server.devicepolicy.DevicePolicyManagerServiceShellCommand.onCommand(DevicePolicyManagerServiceShellCommand.java:87)
        at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
        at android.os.ShellCommand.exec(ShellCommand.java:38)
        at com.android.server.devicepolicy.DevicePolicyManagerService.onShellCommand(DevicePolicyManagerService.java:12290)
        at android.os.Binder.shellCommand(Binder.java:1104)
        at android.os.Binder.onTransact(Binder.java:914)
        at android.app.admin.IDevicePolicyManager$Stub.onTransact(IDevicePolicyManager.java:7077)
        at android.os.Binder.execTransactInternal(Binder.java:1380)
        at android.os.Binder.execTransact(Binder.java:1311)

**Reply by Brokoli5191** [reply]
Try it without "--user 0"

**Reply by extremecarver** [reply]
I did that too already (noticed it sometimes was and sometimes wasn't in the instructions). I'm also not using Samsung Secure folder as that seems to count as an account too. Also I had removed most of the knox stuff via adb, restoring no help. Disabling Secure folder alltogehter - also no help

**Reply by Brokoli5191** [reply]
How about dual messenger?

**Reply by GaneshGorana** [reply]
Just follow installation steps carefully and also hide accounts by installing another app, ther it will get device owner privilege.

**Reply by extremecarver** [reply]
Nope, Dual messenger also not installed or any dual app. Which app to install to hide accounts? The system seems free of accounts as evidenced by pm lister users command and under settings an empty manage accounts overview.

**Reply by GaneshGorana** [reply]
> Nope, Dual messenger also not installed or any dual app. Which app to install to hide accounts? The system seems free of accounts as evidenced by pm lister users command and under settings an empty manage accounts overview.

https://github.com/iamr0s/AndroidAccounts/releases/tag/v1.5
 This app let you copy apps that uses accounts and disable them using hail and also remove other users if there is any and after it just give it device owner permission.

**Reply by Brokoli5191** [reply]
Do it with the account app and follow the instructions above, for me there was a google Service which had an account which wasn't listed in the account settings

**Reply by extremecarver** [reply]
Thanks, that was helpful in finding another app, but still stuck at the same problem (Meetup didn't show up under the settings accounts).

**Reply by extremecarver** [reply]
Woohaa. Reboot again and then it worked. So The full instructions are:

1. install and activate [shizuku](https://shizuku.rikka.app/)
2. install [hail](https://github.com/aistra0528/Hail) and set 'Working mode' to 'Shizuku - Disable'
authorize hail || Alternatively use ADBAppManager disable (both do the same thing)
3. use [Accounts](https://github.com/iamr0s/AndroidAccounts/releases) to get any apps that i needed
import them into hail, and freeze them
4. Remove any SIM Cards/Delete all E-Sim (try without this step maybe it will be fine)
5. Reboot
6. run adb shell dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver on my computer
7. unfreeze the apps, and remove them from the freezing list
8. open hail and set 'Working mode' to 'Dhizuku - Hide' or 'Dhizuku - Suspend'
add apps i need hidden or suspended
9. Done!

**Reply by Brokoli5191** [reply]
Great

**Reply by KimetsuAndrea** [reply]
Bruh it's simple just delete all account In settings> accounts, remove accounts and try again, lol reboot doesn't make sense

**Reply by shockme** [reply]
The reboot was essential to get past the error, after using Hail to hide the apps with accounts.

**Reply by itsmaybetokyo** [reply]
can confirm, it was the reboot what was locking me, Pixel 5

**Comment by extremecarver** [comment]
How can i uninstall and return the rights? Galaxy store cannot update apps or how could I give device owner rights to Galaxy store? Some system apps on Samsung need update via galaxy store and that doesn't work with device owner mode rights belonging to dhizhuku.

**Reply by rajathsingh12** [reply]
Were you able to find any fix to make Galaxy Store the device owner, @extremecarver?

**Reply by extremecarver** [reply]
No, I had to hard reset my phone. So far with the problems  that get created I would advise not to use Dhizuku / device admin on Samsung phones.

**Reply by MARK0W0** [reply]
Ez do a back up and factory rest

**Comment by wlnsjw** [comment]


**Reply by GaneshGorana** [reply]
I didn't understand,  but how i able to delete that app is i disabled the permission within the app, as i remember there was a button to do it, from then my mobile stuck on that screen for 30 minutes,  and somehow my mobile working after that, but that not disabled the permission,  so i removed using adb and restarted my device and then it worked normal

**Comment by Lumintb** [comment]
各位大佬，请问错误显示这样怎么办？
![IMG_20250602_064927](https://github.com/user-attachments/assets/7f25c13d-cd63-42f0-93d4-25530b207277)

**Reply by nktnet1** [reply]
See here:

- https://github.com/BinTianqi/OwnDroid/issues/233#issuecomment-4564641131

**Comment by ia-S-on** [comment]
NEW SOLUTIONS?

For android 14, on T-Mobile Revvl 7

I did:
dpm set-active-admin [com.ro.../...Receiver]
Then:
dpm set-profile-owner [com.ro.../...Receiver]

and it worked.

Subsequently, on brand new Cricket Moto G5g running A14:

accounts:clear (were none)->
Shizuku:rish(termux):$ dpm set-device-owner --user 0 com.rosan.dhizuku/.server.DhizukuDAReceiver

Instant success. so the stock method works, especially when with no accounts, Hail is unnecessary.

Later still, another XT2417,  from Metro, about:phone crashed consistently until updating to A15 finally shizuku running, the stock method failed ("accounts") but the 2 part method from the Revvl 7 worked.

**Reply by femboy-nya** [reply]
Спасибооооо

**Comment by Isma-Andri** [comment]
how to re create the user 999 I just deleted ? Dual apps on Xiaomi needs it

**Comment by Tacofromeverywhere** [comment]


**Reply by wilson2533** [reply]
> Everytime I try to activate dhizuku it says bad admin even when I have removed all accounts and users on my moto g play 2024, is there anything I can do to fix this issue.

Uninstall WhatsApp and Telegram

**Reply by Tacofromeverywhere** [reply]
I don't got those

**Comment by Tacofromeverywhere** [comment]
No receiver above 2.10.1
Device: Moto g play 2024
Android version: 14
Rooted: no

**Comment by MLuqmanBR** [comment]
Help Me! There was no "Dhizuku" option present in the list for me so in confusion I accidentally gave device owner permission to "Hail" but now I am stuck as I am not getting solutions as to what I can do further. Please provide guidance. (Note: I am trying my max not to reset my phone)
![Screenshot_2025_1207_043501-1](https://github.com/user-attachments/assets/6380ad91-73f0-4e00-a11a-3c959070eed3)

**Reply by MLuqmanBR** [reply]
I gave and reset my device and after doing all that work I figured out that there is a button in "Hail" to disable DeviceOwner permission. And as for setting "Dhizuku" as DeviceOwner I installed the older version of the app specifically "2.10.1" and it had that option in the list and so now everything is "working fine" with this older version of the app.

**Comment by iantzzu** [comment]
Hello Sir! I have google pixel 9 pro xl with the latest version of GrapheneOs, with Shizuku and Dhizuku installed! 
I confess that I did not document myself before and curiosity pushed me to test the Dhizuku application. The first attempt led me to success, but in the application grid for "activation by Shizuku" I have two applications: 1. Dhizuku and 2. μLauncher 
After I tried with μLauncher to activate Shizuku, I couldn't do anything and I got the message "Java lang illegal exception..." 
What can I do to reverse this problem? thanks in advance!
![photo_2025-12-08_17-09-04](https://github.com/user-attachments/assets/b8ee21b4-6e6d-4cc0-98df-d9fa3822a952)
![photo_2025-12-08_17-09-14](https://github.com/user-attachments/assets/95453f2d-26d5-4602-8ca5-852749f661f6)

**Reply by iantzzu** [reply]
Now I Can't deactivate this App!
![photo_2025-12-08_17-15-32](https://github.com/user-attachments/assets/f2e27ba4-6630-40c4-8ee0-c11e38994bac)

**Reply by MLuqmanBR** [reply]
Can you show what the launcher's settings show in place of this part now?
<img width="1079" height="646" alt="Screenshot_2025_1208_211215" src="https://github.com/user-attachments/assets/95bbfac1-66c2-4423-a40f-d002a07bd1c2" />

**Reply by iantzzu** [reply]
This option is Blocked. I tried to make a screenshot but is not permited.
I thinking as a Nob that is a command for adb to deactivate μLauncher. if i succeded with aShell.apk

**Reply by MLuqmanBR** [reply]
Then if you don't see any option in the app to deactivate device owner, I think the only option is to reset the device.

**Reply by iantzzu** [reply]
yes! it's an option! thank you so much! apreciate your help.

**Reply by iantzzu** [reply]
maybe the only option!

**Comment by Overhead1387** [comment]
For pixel owners: when you removed all accounts including the Google account. But it still shows the Google account as logged in even after rebooting. It's bec you have private space turned on and you need to delete the private space first. Otherwise you won't be able to logout of your Google entirely.

**Reply by Overhead1387** [reply]
@iamr0s  could you maybe include this in the instructions? Thank you!

**Comment by tefotariq** [comment]
Hello everyone, I am new to Github. I've downloaded some stuff from here before but still haven't gotten the entire understanding of the platform. All i know is that its a place to download stuff. I am currently using a Samsung A24 (Budget Android), and I came across shizuku and dhizuku from reddit threads because i was trying to make it impossible to uninstall important apps on my phone for productivity. I do fall back to bad habits sometimes and delete these my productivity apps  through safe mode. I saw a thread talking about an app called ownDroid, which apparently can help me get access to critical features on my phone to prevent apps from being uninstalled on my phone and make things extremely unbypassible.

 Now before i came across this thread i had tried ways like rooting my device, to turn user apps to system apps on my phone, but still it ddin't work. So i unrooted my device and yes the knox fuse security thingy is tripped so i also lost access to samsung  apps like health. I've looked for other Mobile Device Management software apps to help me improve productivity however they can't work without knox even after reseting my phone to try and install them with their QR codes. So that was a bust and this thread i came across reddit seems promising for me to try.
 
 Now the main issue is I have followed every step to install and activate shizuku, I've used ADB to activate shizuku and it is running. used hail and given it permission using shizuku to freeze all accounts on my phone. I do not share this device with anyone so there aren't any users besides User 0. I've also installed the accounts app to make sure and transfer all accounts into Hail to freeze yet still even if I try to activate Dhizuku either with Shizuku or ADB it does say it is already admin on my device however when opening the Dhizuku app it still says its waiting for activation in RED. Trying to activate it with Shizuku I've selected the app and it does say You are device owner, so i clicked finish and still there is no luck
 
![Screenshot_20260130_102925_Dhizuku 1](https://github.com/user-attachments/assets/abbd9ee4-b8c3-4c31-8bd7-f3059e6bb6e5)
![Screenshot_20260130_102921_Dhizuku 1](https://github.com/user-attachments/assets/d90b9d8d-a36a-4a9b-8041-7c1e29adcb1c)

When trying to activate Dhizuku using ADB, this is what i get back from the command

Exception occurred while executing 'set-device-owner':
java.lang.RuntimeException: Can't set package ComponentInfo{com.rosan.dhizuku/com.rosan.dhizuku.server.DhizukuDAReceiver} as device owner.
at com.android.server.devicepolicy.DevicePolicyManagerServiceShellCommand.runSetDeviceOwner(qb/104304523 f6ef724cebfcd0e5b8c60c4ca08e340007f80c7d693ee790779e4f8001b3f2df:118)
at com.android.server.devicepolicy.DevicePolicyManagerServiceShellCommand.onCommand(qb/104304523 f6ef724cebfcd0e5b8c60c4ca08e340007f80c7d693ee790779e4f8001b3f2df:360)
at com.android.modules.utils.BasicShellCommandHandler.exec(BasicShellCommandHandler.java:97)
at android.os.ShellCommand.exec(ShellCommand.java:38)

Now i am lookingfor a way to go about this before factory reseting my phone and see if its possible to do this even when knox is tripped on my samsung android 16 or would i even be wastinf time even if i did back up my phone after factory reseting.

**Reply by tefotariq** [reply]
Same thing is the case with the ownDroid app, it gives the same error as dhizuku when trying to activate it with ADB and it just says Failed when trying to activate it using Shizuku. In case you are wondering, yes i have given bothe Dhizuku and ownDroid permission from Shizuku after activating

**Reply by nktnet1** [reply]
See if you can apply the solution here to your use case (requires root):

- https://forum.khadas.com/t/the-issue-for-setting-as-device-owner-via-adb-command/15982

I assume the cause is:

- https://github.com/iamr0s/Dhizuku/issues/36#issuecomment-2854547395


---

EDIT:

It could also be:

- https://github.com/BinTianqi/OwnDroid/issues/233#issuecomment-4564641131

**Comment by OussamaTeyib** [comment]
Hi everyone,

Recently, I encountered an issue while trying to activate Dhizuku. I was using the latest version (2.11.2). Even after completing all the required steps and seeing the “You are an owner” message, the app continued to display “Waiting for activation.”

I believe the problem is related to changes introduced in version 2.11 and later. As a workaround, I downgraded to version 2.10.1, which allowed me to activate Dhizuku successfully.

The steps to follow are:
- Set up Shizuku
- Remove or freeze all accounts (or perform a factory reset if this does not help)
- Reboot the device
- Activate Dhizuku using Shizuku

**Comment by namankumar9** [comment]
Before starting, I'd like to know if it's reversible. Can I go back to having no device owner app?

**Reply by nktnet1** [reply]
Yes - there is a disabled button in the UI, as well as a method to transfer device ownership to another application.

**Comment by watermarkhu** [comment]
For Oppo on ColorOS 16, the workaround of disabling apps with accounts via Hail and Shizuku will not work. The number of accounts remains after disabling the apps, which can be checked via adb `adb shell dumpsys account list` (must return 0 accounts). You may get a pop-up with "You are Owner!!", but Dhizuku is never activated. 

In stead, the accounts needs to be manually removed from settings -> Users & Accounts.
